lib/matplotlib/colors.py

```diff
@@ -62,55 +65,201 @@
 import numpy as np
 from numpy import ma
 import matplotlib.cbook as cbook
-from ._color_data import XKCD_COLORS, CSS4_COLORS
+from ._color_data import BASE_COLORS, CSS4_COLORS, XKCD_COLORS
 
-# for back copatibility
-cnames = CSS4_COLORS
 
-COLOR_NAMES = {'xkcd': XKCD_COLORS,
-               'css4': CSS4_COLORS}
+class _ColorMapping(dict):
+    def __init__(self, mapping):
+        super(_ColorMapping, self).__init__(mapping)
+        self._cache = {}
```

Comment:
The cache is needed for performance purposes.  In its absence, `matplotlib.tests.test_lines.test_invisible_Line_rendering` fails.

seaborn needs to reset the cache when changing the named colors mapping (https://github.com/mwaskom/seaborn/blob/dfdd1126626f7ed0fe3737528edecb71346e9eb0/seaborn/palettes.py#L953), most likely because changing just the main mapping would have no effect on already cached conversions.  So I need to maintain it as "semi-public" (via `colorConverter`), but I'd rather make it an implementation detail (thus the `_ColorMapping` class instead).